api/lua/Cargo.toml

```diff
@@ -0,0 +1,80 @@
+[package]
+name = "playdate-lua"
+version = "0.1.0"
+readme = "README.md"
+description = "High-level Lua API built on-top of Playdate API"
+keywords = ["playdate", "sdk", "api", "gamedev"]
+categories = ["game-development", "api-bindings", "no-std"]
+edition.workspace = true
+license.workspace = true
+authors.workspace = true
+homepage.workspace = true
+repository.workspace = true
+
+
+[features]
+default = ["sys/default"]
+
+# playdate-sys features, should be shared because it's build configuration:
+
+bindgen-runtime = ["sys/bindgen-runtime"]
+bindgen-static = ["sys/bindgen-static"]
+bindings-derive-debug = ["sys/bindings-derive-debug"]
+
+
+[dependencies]
+sys = { workspace = true, default-features = false }
+
+[dev-dependencies]
+system = { workspace = true, default-features = false, features = [ "try-trait-v2" ] }
+
+
+[[example]]
+name = "add-function-get-arg-string"
+crate-type = ["dylib", "staticlib"]
+path = "examples/add-function-get-arg-string.rs"
+required-features = ["sys/entry-point", "sys/lang-items"]
+
+
+[package.metadata.playdate]
+bundle-id = "rs.playdate.lua"
+
+[package.metadata.playdate.dev-assets]
+# The Lua runtime expects us to provide a main.pdz file at the root.
+#
+# Compiled with `pdc --skip-unknown --strip sources/main.lua Example.pdx`
+
+# This results in target/playdate/assets/playdate-lua-244d63d426e998ae/dev/build/main.pdz/examples/Example.pdx/main.pdz (not what we want)
+# "main.pdz" = "examples/Example.pdx/main.pdz"
+
+# This results in target/playdate/assets/playdate-lua-244d63d426e998ae/dev/build/examples/Example.pdx/main.pdz (not what we want)
+"/" = "examples/Example.pdx/main.pdz"
```

Comment:
I think either of these would be most convenient but aren't doing what I would expect.

Reply:
Answer in two parts.

### Part first

That is not a bug, that is feature by-design ;)
Probably problematic feature but it is. May be could be explained better.

So, you're set up rule 
```toml
"/" = "examples/Example.pdx/main.pdz"
```

That means "put matched files __with matched paths__ to the root (left part `/`).

As proof you can run command
```bash
cargo playdate package -p=playdate-lua --example=add-function-get-arg-string --features=sys/lang-items,sys/entry-point
```
and then look at assets-build-plan:
```bash
cat target/playdate/assets/playdate-lua-*/playdate-lua-*-dev.json
```

#### How to fix it

The rule should be:
```toml
"/main.pdz" = "examples/Example.pdx/main.pdz"
```
But _part two._


### Part second

🚨 __This IS a bug!__

As I mentioned before, matched part of path is preserved. And for this case this is a problem.
So `"/main.pdz" = "examples/Example.pdx/main.pdz"` puts matched path __into__ the __directory `main.pdz`__ and this is a bug.


### Workaround

This is terrible workaround.

1. put `main.pdz` into the root of the `lua`-crate
2. rule can be any so, for example simply `"main.pdz" = true`

- - -

Okay, thank you. I'll fix it somehow 🤔, it must be fixed.

Reply:
Okay, I know where is problem is. I'll fix it in couple days, I suppose.